azure/services/virtualmachineimages/images.go

```diff
@@ -87,6 +87,9 @@ func (s *Service) GetDefaultWindowsImage(ctx context.Context, location, k8sVersi
 
 	if osAndVersion == "" {
 		osAndVersion = azure.DefaultWindowsOsAndVersion
+		if !k8sVersionInSKUName(v.Major, v.Minor, v.Patch) {
+			osAndVersion += "-containerd"
```

Comment:
I'm not following why we need to add this both here and below. Seems like we're covering both cases for the new and old naming but separately, I'm assuming that's where the bug came from?

Reply:
I think if we just move 

```
if v.GTE(v122) && runtime != "dockershim" {
		osAndVersion += "-containerd"
	}
```

 up to this line it should cover both cases before and after the rename

Reply:
Ha! I pushed the same change a bit ago. I'm also checking that the `osAndVersion` doesn't already end with that suffix.